src/storage/txn/scheduler.rs

```diff
@@ -437,15 +437,40 @@ impl<E: Engine, L: LockManager, P: PdClient + 'static> Scheduler<E, L, P> {
         SCHED_STAGE_COUNTER_VEC.get(tag).read_finish.inc();
 
         debug!("read command finished"; "cid" => cid);
-        let tctx = self.inner.dequeue_task_context(cid);
-        if let ProcessResult::NextCommand { cmd } = pr {
+        if let ProcessResult::NextCommands { cmds } = pr {
+            let cmd_count = cmds.len();
+            let mut tctx = self.inner.dequeue_task_context(cid);
             SCHED_STAGE_COUNTER_VEC.get(tag).next_cmd.inc();
-            self.schedule_command(cmd, tctx.cb.unwrap());
+            let cb = tctx.cb.take();
+            let (tx, rx) = std::sync::mpsc::channel();
```

Comment:
A std sync channel can block its thread. So when running the future below, we'll have one less running threads in this scheduler pool. We should use an async channel.